pages/_app.js

```diff
@@ -1,10 +1,43 @@
 import GlobalStyle from "../styles";
+import useLocalStorageState from "use-local-storage-state";
+
+import activities from "@/assets/activities";
 
 export default function App({ Component, pageProps }) {
+  const initialActivities = activities.map((activity) => ({
+    ...activity,
+    id: activity.id,
```

Comment:
why do you overwrite the id with the same id?
But the point is: Please do not duplicate your source of truth. This will produce lots of bugs.  Either expand your activities with the isBookmarked prop OR store only additional information (the isBookmarked state) in the activitiesInfo.